examples/tinywl-new/wallpaperimage.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (C) 2024 UnionTech Software Technology Co., Ltd.
+// SPDX-License-Identifier: Apache-2.0 OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+
+#include "wallpaperprovider.h"
+#include "wallpaperimage.h"
+#include "helper.h"
+#include "output.h"
+#include "woutput.h"
+#include "workspace.h"
+
+#include <QDir>
+
+WallpaperImage::WallpaperImage(QQuickItem *parent)
+    : QQuickImage(parent)
+{
+    auto provider = Helper::instance()->qmlEngine()->wallpaperImageProvider();
+    connect(provider, &WallpaperImageProvider::wallpaperTextureUpdate, this, &WallpaperImage::updateWallpaperTexture);
+
+    auto workspace = Helper::instance()->workspace();
+    connect(workspace, &Workspace::currentChanged, this, &WallpaperImage::updateSource);
+}
+
+WallpaperImage::~WallpaperImage()
+{
+
+}
+
+int WallpaperImage::userId()
+{
+    return m_userId;
+}
+
+void WallpaperImage::setUserId(const int id)
+{
+    if (m_userId != id) {
+        m_userId = id;
+        Q_EMIT outputChanged();
+        updateSource();
+    }
+}
+
+WOutput* WallpaperImage::output()
+{
+    return m_output;
+}
+
+void WallpaperImage::setOutput(WOutput* output)
+{
+    if (m_output != output) {
+        m_output = output;
+        Q_EMIT outputChanged();
+        updateSource();
+    }
+}
+
+void WallpaperImage::updateSource()
+{
+    auto workspace = Helper::instance()->workspace();
+    if (m_userId == -1 ||
+        !m_output ||
+        !workspace) {
+        return;
+    }
+
+    QString home = QStandardPaths::writableLocation(QStandardPaths::HomeLocation);
```

Comment:
WallpaperImage don't read local file, always use `image://wallpaper/xxxx` url to get wallpaper texture from WallpaperProvider.